### DISCUSSION TITLE
Custom job id

### DISCUSSION THREAD

**Discussion by siavashg87** [taskforcesh/bullmq#669] [General]
Is there a way to clear out all previous custom job ids from the queue?

It does not seem like queue.empty() removes all previous job ids, as my new created jobs with custom ids sometimes gets ignored.

**Comment by manast** [comment]
There is no #empty method in BullMQ, maybe you mean #drain? and as the name implies it only removes the jobs that are currently enqueued, but jobs that have completed or failed are not affected. You can remove those by configuring the "onRemoveComplete/Fail" options: https://docs.bullmq.io/guide/jobs/fifo#default-job-options

**Reply by siavashg87** [reply]
Seems to me that the method exists: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/bull/v2/index.d.ts#L211

However it does not clear out the completed jobs.

Writing this for those who might have this issue in future.

My usecase was that some of my jobs was ignored, due to duplicated job ids, even after completion.

However I managed to remove completed jobs using "clean" method.

I'm aware of "removeOnComplete" option. I would suggest to add a delay/timeout to the option.

**Reply by manast** [reply]
You are referring to a different version of the library, you are on the BullMQ repo, there are no external typings they are provided in the same module since it is written in typescript. The "empty" method in Bull was renamed to "drain" to make it more clear.
